utils/node-loader/src/batch_pool.rs

```diff
@@ -1,373 +1,255 @@
-
-            let (ex_results, batch_block_hash) = api.create_program_bytes_batch(args).await?;
-
-            let mut init_messages = BTreeMap::new();
-
-            for r in ex_results {
-                match r {
-                    Ok((mid, pid)) => {
-                        init_messages.insert(mid, pid);
-                    }
-                    Err(e) => logs.push(format!(
-                        "[#{:<2}] Extrinsic failure: '{:?}'",
-                        logs.len() + 1,
-                        e
-                    )),
-                }
-            }
-
-            let results: Result<Vec<(MessageId, Option<String>)>>;
+            let (ex_results, block_hash) = batch_sender.create_program_batch(args).await?;
+            let messages = process_ex_results(ex_results);
+            process_events(batch_sender.into_gear_api(), messages, block_hash, true).await
+        }
+    }
+}
 
-            let now = utils::now();
+fn process_ex_results<Key: Ord, Value>(
+    ex_results: impl IntoIterator<Item = GClientResult<(Key, Value)>>,
+) -> BTreeMap<Key, (Value, CallId)> {
+    let mut res = BTreeMap::<Key, (Value, CallId)>::new();
+
+    for (i, r) in ex_results.into_iter().enumerate() {
+        let call_id = i + 1;
+        match r {
+            Ok((key, value)) => {
+                res.insert(key, (value, call_id));
+                tracing::debug!("[Call with id: {call_id}]: Successfully executed.")
+            }
+            Err(e) => tracing::debug!("[Call with id: {call_id}]: Failed: '{e:?}'"),
+        }
+    }
 
-            loop {
-                let r = match api.events_since(batch_block_hash, 10).await {
-                    Ok(mut v) => v.err_or_succeed_batch(init_messages.keys().cloned()).await,
-                    Err(e) => Err(e),
-                };
+    res
+}
 
-                if utils::now() - now > 1100 {
-                    results = Err(anyhow!("Out of time: probably blocks stopped.").into());
-                    break;
-                }
+async fn process_events(
+    api: GearApi,
+    mut messages: BTreeMap<MessageId, (ProgramId, usize)>,
+    block_hash: H256,
+    collect_programs: bool,
+) -> Result<Report> {
+    let results: GClientResult<Vec<(MessageId, Option<String>)>>;
+    let now = utils::now();
+    // States what amount of blocks we should wait for taking all the events about successful `messages` execution
+    let wait_for_events_blocks = 10;
+
+    loop {
+        let r = match api.events_since(block_hash, wait_for_events_blocks).await {
+            Ok(mut v) => v.err_or_succeed_batch(messages.keys().copied()).await,
+            Err(e) => Err(e),
+        };
+
+        // If one block is considered to be produced in 1 second, than we wait for 10000 millis (1 sec) * `wait_for_events_blocks`
+        // We also multiply it on the 5 just to be 100% sure if no events occurred, than node is crashed
+        if (utils::now() - now) as usize > wait_for_events_blocks * 1000 * 5 {
```

Comment:
block time should be caught from the runtime

Reply:
done